src/map/script.cpp

```diff
@@ -27340,6 +27340,53 @@ BUILDIN_FUNC(setdialogpospercent){
 	return SCRIPT_CMD_SUCCESS;
 }
 
+/**
+ * Checks if the character has that permission.
+ * checkperm(<permission>{,<char_id>}) -> <bool>
+ */
+BUILDIN_FUNC(checkperm)
+{
+	map_session_data* sd = nullptr;
+
+	if (!script_charid2sd(3, sd))
+		return SCRIPT_CMD_FAILURE;
+
```

Comment:
Please store the permission in a variable.
Then verify the range. (Out of range => script cmd failure)
Then cast it.